libs/transactions/src/lib/directives/transaction-additional-details.directive.ts

```diff
@@ -1,7 +1,15 @@
 import { Directive, TemplateRef } from "@angular/core";
+import { TransactionItem } from "@backbase/data-ang/transactions";
 
+export type AdditionalDetailsContext = {
+  $implicit: TransactionItem
+}
 
 @Directive({selector: 'ng-template[bbTransactionAdditionalDetails]'})
 export class TransactionAdditionalDetailsTemplateDirective {
-  constructor(public templateRef: TemplateRef<any>) {}
+  constructor(public templateRef: TemplateRef<AdditionalDetailsContext>) {}
+
+  static ngTemplateContextGuard(dir: TransactionAdditionalDetailsTemplateDirective, ctx: unknown): ctx is AdditionalDetailsContext { 
```

Comment:
this adds autocomplete / compile time errors for developers in the template when implementing this  extension

if we think this is overkill (I don't) we could just type the templateRef with `any` as the generic

Reply:
I like this, it makes it easy to detect breaking changes to the public APIs

Reply:
I would suggest to add a simple test to the example